addons/website_slides/static/src/interactions/slides_course_join.js

```diff
@@ -1,175 +1,105 @@
+        });
+        this.user = this.slidesService.data.user;
+        this.channel = this.slidesService.data.channel;
+    }
 
     /**
-     *
-     * Overridden to add options parameters.
-     *
-     * @param {Object} parent
-     * @param {Object} options
-     * @param {Object} options.channel slide.channel information
-     * @param {boolean} options.isMember whether current user is enrolled
-     * @param {boolean} options.isMemberOrInvited whether current user is at least invited
-     * @param {string} options.inviteHash hash of the invited attendee. Needed to grant
-     *   access to a course preview / to identify.
-     * @param {integer} options.invitePartnerId id of partner of invited attendee if any.
-     *   Also needed to access course preview / to identify.
-     * @param {boolean} options.invitePreview whether the course is rendered as a preview.
-     *   This is true when an invited attendee is on the course while unlogged.
-     * @param {boolean} options.isPartnerWithoutUser whether invited partner has users. Used
-     *   to redirect properly to sign up / log in.
-     * @param {string} [options.joinMessage] the message to use for the simple join case
-     *   when the course is free and the user is logged in, defaults to "Join this Course".
-     * @param {Promise} [options.beforeJoin] a promise to execute before we redirect to
-     *   another url within the join process (login / buy course / ...)
-     * @param {function} [options.afterJoin] a callback function called after the user has
-     *   joined the course
+     * @param {MouseEvent} event
      */
-    init: function (parent, options) {
-        this._super.apply(this, arguments);
-        this.channel = options.channel;
-        this.isMember = options.isMember;
-        this.isMemberOrInvited = options.isMemberOrInvited;
-        this.inviteHash = options.inviteHash;
-        this.invitePartnerId = options.invitePartnerId;
-        this.invitePreview = options.invitePreview;
-        this.isPartnerWithoutUser = options.isPartnerWithoutUser;
-        this.publicUser = options.publicUser;
-        this.joinMessage = options.joinMessage || _t("Join this Course");
-        this.beforeJoin =
-            options.beforeJoin ||
-            function () {
-                return Promise.resolve();
-            };
-        this.afterJoin =
-            options.afterJoin ||
-            function () {
-                document.location.reload();
-            };
-    },
-
-    async onJoinClick(ev) {
-        ev.preventDefault();
-
+    async onJoinClick() {
         if (
-            this.invitePreview ||
-            (this.channel.channelEnroll === "invite" && this.isMemberOrInvited)
+            this.channel.invitePreview ||
+            (this.channel.enroll === "invite" && this.channel.isMemberOrInvited)
         ) {
-            this.joinChannel(this.channel.channelId);
+            this.joinChannel(this.channel.id);
             return;
```

Comment:
Shouldn't return the promise directly (and await below)?
```suggestion
            return this.joinChannel(this.channel.id);